libs/@blockprotocol/service/src/scripts/generate-service-module-json.ts

```diff
@@ -0,0 +1,232 @@
+import { writeFileSync } from "node:fs";
+/* eslint-disable import/no-extraneous-dependencies */
+import { resolve } from "node:path";
+
+import { generateSchema, getProgramFromFiles } from "typescript-json-schema";
+
+export type ServiceJsonMessage = {
+  messageName: string;
+  description: string;
+  source: "block" | "embedder";
+  respondedToBy: string | null;
+  sentOnInitialization: boolean;
+  data: any;
+  errorCodes: string[];
+};
```

Comment:
Could this be `ModuleMessageDefinition` from `@blockprotocol/core`, or based on it?

Reply:
Ah wasn't aware this was available, have made use of `ServiceMessageDefinition` in https://github.com/blockprotocol/blockprotocol/pull/954/commits/0a9e5b7dd87ca18fc14b5cddd0f8610d01408c65